wallet/txauthor/author.go

```diff
@@ -154,6 +154,49 @@ func NewUnsignedTransaction(outputs []*wire.TxOut, relayFeePerKb dcrutil.Amount,
 	}
 }
 
+// NewUnsignedTransactionMinusFee creates an unsigned transaction paying to one
+// non-change output.  An appropriate transaction fee is included based on the
+// transaction size, and is subtracted from the provided `output` parameter.
+//
+// The behavior of this method mirrors a call to `NewUnsignedTransaction` with the differences being that:
+// * this method takes a single `output` argument
+// * all fees are subtracted from the provided output rather than from change.
+// * the output in the return value is a shallow copy of the output argument and references the original output script.
+//
+// `NewUnsignedTransaction` only performs dust checks on the change output. As a result, this method does not perform
+// dust checks due to the lack of a change output.
+func NewUnsignedTransactionMinusFee(output *wire.TxOut, relayFeePerKb dcrutil.Amount,
+	fetchInputs InputSource, fetchChange ChangeSource) (*AuthoredTx, error) {
```

Comment:
This API needs to be revisited if we are exposing a new API here instead of just creating a wrapper over NewUnsignedTransaction (since that can be done by any other package).  We should drop the fetchChange parameter since the purpose of this function is to never create change.

Reply:
"... the purpose of this function is to never create change". If there should not be change, it seems like the implied utility is to sweep an account, but the expectation is not clear. What is the expected behavior for a single input that exceeds the size of the output by more than dust?

Reply:
oh derp, you're right. forget that, we do still need to create change.

Reply:
To be fair, this API has room for expansion or being promoted to the caller as a private method and expanded later.  If the change is dust, the user might want to specify how to handle that edge case.

if the change is dust...
1. return error in the scenario where the recipient must pay all fees (GiveDust=false & IncreaseRelayFee=false)
2. subtract dust-change from the fee paid by the recipient (GiveDust=true & IncreaseRelayFee=false).
3. add dust-change to the fee paid by the recipient (GiveDust=true & IncreaseRelayFee=true)
4. GiveDust=false & IncreaseRelayFee=true ??? out of scope for this method, error. or deprecate the existing method and include RecipientPaysFee in this new options struct.

From the cli, I'm thinking the default should be the most restrictive (false & false). Whatever we default to, we need to stick with. Moving to the caller and going private would let us defer on this. Thoughts?